### PR TITLE
refactor: convert Message to interface

### PR DESCRIPTION
**Pull request by jcosentino11**
**Issue #, if available:**

**Description of changes:**
Convert `Message` from a class to an interface, with the ability for it to convert itself to mqtt or pubsub types.

**Why is this change necessary:**
To keep `Message` generic without cluttering it with Mqtt5 fields, which will be added in the near future

**How was this change tested:**
unit tests

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/aws/greengrass/mqtt/bridge/model/Message.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+public interface Message {
+    Message toPubSub();
+
+    Message toMqtt();
+
+    String getTopic();
+
+    byte[] getPayload();
+
+    Message ofTopic(String topic);
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/mqtt/bridge/model/Message.java:20]: why does this exist? We already have a `Message`, so why are we then calling a method on it to give us a different message which a presumably different topic?

File: src/main/java/com/aws/greengrass/mqtt/bridge/model/MqttMessage.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+package com.aws.greengrass.mqtt.bridge.model;
+
+import lombok.NonNull;
+import lombok.Value;
+
+// TODO support mqtt5
+@Value
+public class MqttMessage implements Message {
```
Comment (MikeDombo) [inline src/main/java/com/aws/greengrass/mqtt/bridge/model/MqttMessage.java:13]: Rename to LocalMqttMessage

File: src/main/java/com/aws/greengrass/mqtt/bridge/model/Message.java (hunk 1 above)
Comment (jcosentino11) [reply src/main/java/com/aws/greengrass/mqtt/bridge/model/Message.java:20]: Yeah, cause of generics weirdness https://github.com/aws-greengrass/aws-greengrass-mqtt-bridge/pull/88/files#diff-7d623726e13974aaebfada082983f1677688fcc2f9225dc9a51c63a580ee89c9R145-R150.  I guess could just make it a setTopic, but that would remove the immutability of message

File: src/main/java/com/aws/greengrass/mqtt/bridge/model/MqttMessage.java (hunk 2 above)
Comment (jcosentino11) [reply src/main/java/com/aws/greengrass/mqtt/bridge/model/MqttMessage.java:13]: Was also using this in iot core client, did we want a separate type for that?

File: src/main/java/com/aws/greengrass/mqtt/bridge/model/Message.java (hunk 1 above)
Comment (MikeDombo) [reply src/main/java/com/aws/greengrass/mqtt/bridge/model/Message.java:20]: I think the name is confusing. How about `newFromMessageWithTopic`?

Comment (jcosentino11) [reply src/main/java/com/aws/greengrass/mqtt/bridge/model/Message.java:20]: yeah will change

Comment (jcosentino11) [reply src/main/java/com/aws/greengrass/mqtt/bridge/model/Message.java]: done

**Review by MikeDombo — APPROVED** [review]

**Review by robcmann — APPROVED** [review]

**Pull request merged**